src/main/java/uk/nhs/digital/nhsconnect/nhais/model/fhir/ParametersExtension.java

```diff
@@ -0,0 +1,69 @@
+package uk.nhs.digital.nhsconnect.nhais.model.fhir;
+
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
+import uk.nhs.digital.nhsconnect.nhais.exceptions.FhirValidationException;
+
+import org.hl7.fhir.r4.model.Parameters;
+import org.hl7.fhir.r4.model.Patient;
+import org.hl7.fhir.r4.model.Resource;
+import org.hl7.fhir.r4.model.StringType;
+import uk.nhs.digital.nhsconnect.nhais.model.edifact.AcceptanceType;
+
+@RequiredArgsConstructor
+public class ParametersExtension {
+
+    private final Parameters parameters;
+
+    public static Patient extractPatient(Parameters parameters) {
+        return new ParametersExtension(parameters).extractResource(ParameterNames.PATIENT, Patient.class);
+    }
+
+    public Patient extractPatient() {
```

Comment:
for future use or not needed anymore?

Reply:
this one is not static - the static one is actually calling it

Reply:
or at least it should :)

Reply:
will be used in @grzegorz-robak  PR

Reply:
Yeah I just fixed that :D